https/gifts.js

```diff
@@ -5,113 +5,56 @@ also has functions to return admin stuff as well
 */
 const functions = require('firebase-functions');
 require('firebase/firestore');
-const { getUUID, getGeneralQueries } = require('../utils/utils');
-const { YEAR } = require("../config/constants");
+const { getUUID, getGeneralQueries, markGifteeAccount } = require('../utils/utils');
+const { EVENT } = require("../config/constants");
 const db = require('../config/db');
 
 // marks gift as went
-const sendGift = functions.https.onCall(async({user}, context) => {
-  // marks their own account
+const sendGift = functions.https.onCall(async({user, value, giftee_uuid}, context) => {
+  // this has to mark both the giftee and gifter
 
+  if(typeof value === "undefined"){value = true}
+  // gifter first
   let uuid = await getUUID(user)
   if(uuid.error){return {error: "no API key set"}}
-  uuid = uuid.success
+  let entryResult = await db.collection('events').doc(EVENT).collection('participants').doc(uuid.success).set({ sent_own: value }, {merge: true}).then(()=> {return true}).catch(() => {return false});
 
-
-  let entryResult = await db.collection('events').doc(YEAR).collection('participants').doc(uuid).set({ sent: true }, {merge: true}).then(()=> {return true}).catch(() => {return false});
+  // giftee now, the giftee's uuid is known
+  let gifteeStatus = await markGifteeAccount({uuid:giftee_uuid}, { field: "sent", value:value })
```

Comment:
Could `giftee_uuid` be null?

Reply:
Good point, but if it were I would blame the frontend. Basically a gifter can have multiple giftees and so the uuid is required to find the specific one

Reply:
Alright, we'll make sure to have some checks on frontend as well.